instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractor.java

```diff
@@ -64,7 +64,11 @@ private static String stripSensitiveData(@Nullable String url) {
     }
 
     // replace username & password
-    return url.substring(0, url.indexOf("//")+2) + url.substring(url.indexOf("@")+1);
+    if(url.contains("@")) {
+      return url.substring(0, url.indexOf("//")+2) + url.substring(url.indexOf("@")+1);
```

Comment:
Nit: no need to lookup `"@"` twice:
```suggestion
    int atIndex = url.indexOf("@");
    if (atIndex != -1) {
      return url.substring(0, url.indexOf("//")+2) + url.substring(atIndex+1);
```